manifest/pack_pull_git.go

```diff
@@ -188,7 +188,30 @@ func pullGit(ctx context.Context, manifest *Manifest, opts ...pack.PullOption) e
 		WithField("branch", popts.Version()).
 		Infof("git clone")
 
-	if _, err = git.PlainCloneContext(ctx, local, false, copts); err != nil {
+	_, err = git.PlainCloneContext(ctx, local, false, copts)
+	switch err {
+	case git.ErrRepositoryAlreadyExists:
+		reps, err := git.PlainOpen(local)
+		if err != nil {
+			return fmt.Errorf("could not open repository: %w", err)
+		}
+		err = reps.FetchContext(ctx, &git.FetchOptions{
+			RemoteURL: copts.URL,
+			Tags:      copts.Tags,
+			Depth:     copts.Depth,
+			Auth:      copts.Auth,
+			Progress:  nil,
+		})
+		switch err {
+		case git.NoErrAlreadyUpToDate, git.ErrBranchExists, nil:
+			log.G(ctx).Infof("successfully updated %s in %s", path, local)
+			return nil
+		default:
+			return fmt.Errorf("could not clone repository: %w", err)
+		}
+	case nil:
+		break
+	default:
 		return fmt.Errorf("could not clone repository: %w", err)
```

Comment:
We only need one case here:

```suggestion
	case err != nil:
		return fmt.Errorf("could not clone repository: %w", err)
```

It it's `nil`, we "break" out of the switch implicitly.

Reply:
you sure know your switches, updated